### PR TITLE
Improve the configuration of excluded User Agents

### LINKED ISSUE #7978: Excluded User Agent still tracked
I had configured the following string under the Websites -> Global list of user agents to exclude:

> "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20100101 Firefox/10.0"

image

It belongs to the Monitor.us service.

Yet later I still saw that bot in the Real-time Visitors:

image

Do I maybe need to add the user agent string without quotes? By the way, there's no explanation of the format in which UA strings should be entered in that box. Presumably, one per line, but who knows? CSV format? Quoted or not?

### PR DESCRIPTION
**Pull request by mnapoli**
Fixes #7978, replaces #8050 

Problems:
- _Excluded User Agents_ is a comma-separated list, but user agents can contain commas
- we want to let users wrap user agents in quotes (see https://github.com/piwik/piwik/issues/7978#issuecomment-121509999) which is currently not supported
- the inline help for _Excluded User Agents_ doesn't say how to format user agent strings (#7978)
- other fields in the page ask for one item per line:
  - it's not consistent with _Excluded User Agents_ (which is comma-separated), so it's confusing
  - in the code it's actually comma-separated, so it's clearly misleading and broken for some users

These problems apply to global and site-specific settings.

Goals:
- [x] consistent formatting for all fields (and both global and site-specific settings)
  - [x] excluded user agents
  - [x] excluded query parameters
  - [x] excluded IPs
- [x] it needs to be "one per line" because of user agents (that can contain commas)
- [x] enclosing quotes need to be accepted (and trimmed) for user agents
- [x] field helps need to match the actual behavior
- [x] migration script
- [x] tests
- [x] update changelog regarding API changes

Notes:
- enclosing quotes for user agents are trimmed (for the other fields I didn't think it would make sense)
- the help text for user agents has also been updated (only the english version), it now explicitly says that user agents should be separated by new lines. What is the best course of action to ensure translations are updated? New translation key?
- The following API methods are now returning an array of strings instead of a list encoded in CSV:
  - `SitesManager.getExcludedIpsGlobal`
  - `SitesManager.getExcludedQueryParametersGlobal`
  - `SitesManager.getExcludedUserAgentsGlobal`
- The following API methods now accept values separated by new lines instead of commas:
  - `SitesManager.setGlobalExcludedIps`
  - `SitesManager.setGlobalExcludedQueryParameters`
  - `SitesManager.setGlobalExcludedUserAgents`

### CODE REVIEW

**Comment by mnapoli** [comment]
Tests are OK, the only fails are for UI screenshots where the translation has been changed: http://builds-artifacts.piwik.org/piwik/piwik/bug-7978/14619/

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 This is a changelog for Piwik platform developers. All changes for our HTTP API's, Plugins, Themes, etc will be listed here.
 
+## Piwik 2.15.0
+
+### Breaking Changes
+* The following API methods are now returning an array of strings instead of a list encoded in CSV:
```
Comment (tsteur) [inline CHANGELOG.md]: Is there maybe an easy way to keep BC?

**Comment by tsteur** [comment]
Is this PR similar to #8368 ? Also needs a rebase.

Re HTTP API break maybe @mattab can have a look

**Comment by mattab** [comment]
@mnapoli can you target this PR for 3.0 branch? we will break the API there and keep 2.15.0 without API breakage :+1:

**Comment by mnapoli** [comment]
> Is this PR similar to #8368?

Not really it's a more global refactoring and improvement to fix 4 different issues (see the PR description).

Will target 3.0.

**Comment by tsteur** [comment]
So we still need to merge #8368 on top?

**Comment by mnapoli** [comment]
No, this issue fixes #7978.

**Comment by mnapoli** [comment]
Closed in favor of #8586 that is targetting 3.0.

**Pull request closed without merging**